---
title: |
  | Supplementary Materials for:
  | Ecosystem size and complexity dictate riverine biodiversity
author:
- Akira Terui^[Department of Biology, University of North Carolina at Greensboro, Greensboro, NC 27412, USA]
- Seoghyun Kim\footnotemark[1]
- Christine L. Dolph^[Department of Ecology, Evolution and Behavior, University of Minnesota, St. Paul, MN 55108, USA]
- Taku Kadoya^[National Institute for Environmental Studies, Tsukuba, Ibaraki 305-8506, Japan]
- Yusuke Miyazaki^[Shiraume Gakuen College, Kodaira, Tokyo 187-8570, Japan]
output:
  pdf_document:
    latex_engine: xelatex
    toc: yes
    toc_depth: 3
bibliography: references.bib
csl: science.csl
knit: (function(inputFile, encoding) {
        rmarkdown::render(inputFile, encoding = encoding, output_dir = "document_output")
      })
---

```{r setup, include=FALSE}

  pacman::p_load(tidyverse, knitr, kableExtra)
  knitr::opts_chunk$set(echo = FALSE,
                        message = FALSE,
                        warning = FALSE)
  
  pacman::p_load(tidyverse, sf)
  
  ## read gis files
  filename <- list.files(path = here::here('empirical/data_gis'), full.names = T)
  wsd_subset <- lapply(filename[str_detect(filename, "wsd_subset")], st_read)
  point_subset <- lapply(filename[str_detect(filename, "point_subset")], st_read)
  shape <- lapply(filename[str_detect(filename, "shape")], st_read)
  shape[[1]] <- st_set_crs(shape[[1]], st_crs(wsd_subset[[1]]))
  
  ## read theory file
  dat_sim <- read_csv(here::here('theory/result/result_sim2020-11-10.csv'))
  
  n_scenario <- dat_sim %>% distinct(n_species,
                       mean_env_source,
                       sd_env_source,
                       rho,
                       asymmetry_factor,
                       min_optim,
                       max_optim,
                       sd_env_lon,
                       theta,
                       K0,
                       sd_env,
                       spatial_env_cor,
                       phi,
                       min_niche_width,
                       max_niche_width,
                       niche_cost,
                       p_dispersal,
                       min_alpha,
                       max_alpha) %>% 
    nrow()
  
  # Hokkaido data
  dat_hkd <- read_csv(here::here("empirical/data_out/data_hkd.csv")) %>% mutate(region = "hokkaido")
  dat_hkd_site <- read_csv(here::here('empirical/data_out/data_hkd_site.csv'))
  
  # Midwest data
  dat_mw <- read_csv(here::here("empirical/data_out/data_mw.csv")) %>% mutate(region = "midwest")
  dat_mw_site <- read_csv(here::here('empirical/data_out/data_mw_site.csv')) %>% 
    mutate(Year = format(as.Date(Date, format = '%m/%d/%Y'), "%Y"))
  
  # tables
  
  ## sensitivity
  source(here::here('theory/table_sensitivity_analysis.R'))

```

# Theory

## Sensitivity analysis

We performed a sensitivity analysis of the metacommunity simulation to identify key simulation parameters that strongly affect the relationships between diversity metrics ($\alpha$, $\beta$, and $\gamma$) and ecosystem properties (the number of habitat patches $N_{p}$ and branching probability $P_{b}$). We generated 500 sets of parameter combinations by randomly drawing values of 8 simulation parameters from uniform distributions (**Table S1**). For each parameter combination, we generated 100 branching networks with the gradients of ecosystem size ($N_p \sim Unif(10, 150)$) and complexity ($P_b \sim Unif(0.01, 0.99)$). This results in a total of 50000 simulation replicates. In each simulation replicate, we allowed interspecific variation in niche optimum $\mu_i$ and width $\sigma_{niche,i}$ ($\mu_i \sim Unif(-1, 1)$ and $\sigma_{niche,i} \sim Unif(0.1, 1)$, respectively; subscript $i$ represents species) and ran 1400 time steps of metacommunity dynamics. We obtained temporal means of $\alpha$, $\beta$, and $\gamma$ diversity for the last 1000 time steps. The first 400 time steps were discarded as initialization and burn-in periods. We removed simulation replicates in which no species established populations over the initial 400 time steps.

For each parameter combination, we regressed log-transformed $\alpha$, $\beta$, and $\gamma$ diversity ($log_{10}~y_j$ for network replicate $j$) on the number of habitat patches $N_{p}$ and branching probability $P_{b}$ as:

```{=tex}
\begin{equation}
\begin{aligned}
  log_{10}~y_j &\sim Normal(\mu_j, \sigma^2)\\
         \mu_j &= \psi_0 + \psi_1~log_{10}~N_{p,j} + \psi_2~log_{10}~P_{b,j}
\end{aligned}
\end{equation}
```
where $\psi_k$ ($k = 0-2$) are the intercept ($\psi_0$) and regression coefficients ($\psi_1$ and $\psi_2$). We extracted 500 estimates of $\psi_1$ and $\psi_2$, which represent the effects of $N_{p}$ and $P_{b}$ on diversity metrics under a given parameter combination. To examine influences of simulation parameters (**Table S1**) on $\psi_1$ and $\psi_2$, we developed the following regression model taking $\psi_1$ or $\psi_2$ as a response variable $u_n$ (parameter combination $n$):

```{=tex}
\begin{equation}
\begin{aligned}
  u_{n} &\sim Normal(\mu_n, \sigma^2)\\
  \mu_n &= \zeta_0 + \zeta_1 \sigma_{h,n} + \zeta_2 \sigma_{l,n} + \zeta_3 \sigma_{z,n} + \zeta_4 \phi_n + \zeta_5 \nu_n + \zeta_6 \alpha_{max, n} + \zeta_7 \theta_n + \zeta_8 p_{d,n} 
\end{aligned}
\end{equation}
```
where $\zeta_k$ ($k = 0-8$) are the intercept ($\zeta_0$) and regression coefficients ($\zeta_{1-8}$). Explanatory variables were standardized to a mean of zero and a standard deviation of one, so that regression coefficients are comparable.

The sensitivity analysis revealed key simulation parameters. For the effects of $N_p$, the following simulation parameters were influential: the degree of local environmental noise ($\sigma_l$; influenced the effects on $\alpha$ and $\gamma$ diversity), the maximum value of interspecific competition coefficient ($\alpha_{max}$; influenced the effects on $\alpha$, $\beta$, and $\gamma$ diversity), dispersal distance ($\theta$; influenced the effects on $\alpha$ and $\beta$ diversity), and dispersal probability ($p_d$; influenced the effect on $\alpha$ diversity) (**Table S3**). For the effects of $P_b$, the following simulation parameters were influential: environmental variation at headwaters ($\sigma_h$; influenced the effect on $\gamma$ diversity), the degree of local environmental noise ($\sigma_l$; influenced the effects on $\alpha$ and $\beta$ diversity), and dispersal distance ($\theta$; influenced the effects on $\alpha$ and $\beta$ diversity) (**Table S4**).

Based on the results, we identified $\sigma_h$, $\sigma_l$, $\alpha_{max}$, $\theta$, and $p_d$ as key parameters. We changed the values of these parameters in the main analysis and examined the relationships between diversity metrics and ecosystem properties.

## Longitudinal gradient of local species richness

Longitudinal gradients of local species richness have been extensively studied in rivers, illuminating typical patterns observed in nature. The most common pattern is a downstream increase of local species richness [@mcdowallFightingFlowDownstreamupstream1998; @rahelFishAssemblagesHabitat1991; @teruiThreeEcologicalFactors2016]. However, recent empirical and theoretical studies also showed 'reversed patterns,' in which local species richness decreases downstream [@bsemerHeadwatersAreCritical2013; @harveyDisturbanceReversesClassic2018]. We predicted the longitudinal gradient of local species richness to confirm that our simulation scenarios are capable of reproducing the previously observed patterns of local species richness. We considered 32 simulation scenarios comprising four landscape and eight ecological scenarios, as described in the main text (a set of parameters is described in **Table S2**). Under each simulation scenario, we generated 10 branching networks with fixed parameters of ecosystem size ($N_p = 100$) and complexity ($P_b = 0.5$). This results in a total of 320 simulation replicates. In each simulation replicate, we allowed interspecific variation in niche optimum $\mu_i$ and width $\sigma_{niche,i}$ ($\mu_i \sim Unif(-1, 1)$ and $\sigma_{niche,i} \sim Unif(0.1, 1)$, respectively; subscript $i$ represents species) and ran 1400 time steps of metacommunity dynamics. We obtained temporal means of local species richness at each habitat patch for the last 1000 time steps. The first 400 time steps were discarded as initialization and burn-in periods. We removed simulation replicates in which no species established populations over the initial 400 time steps. We evaluated the relationship between local species richness and the number of upstream habitat patches, a proxy for the longitudinal position of a habitat patch.

The simulation reproduced diverse patterns of longitudinal gradients in local species richness (**Figures S1-4**). The downstream increase of local species richness was predicted under a natural landscape scenario, in which environmental variation at headwaters $\sigma_h$ exceeds the degree of local environmental noise $\sigma_l$ (**Figure S1**). This pattern was consistent across ecological scenarios except those with long dispersal distance and high dispersal probability (**Figure S1**). Similarly, we observed a downstream increase of local species richness in scenarios with low habitat diversity ($\sigma_h = \sigma_l = 0.01$) and low dispersal probability ($p_d = 0.01$) (**Figure S3**). However, there were cases where local species richness decreased downstream or showed no longitudinal patterns. For example, when local environmental noise exceeds environmental variation at headwaters ($\sigma_l \ge \sigma_h$), local species richness showed a downstream decrease or a vague longitudinal pattern (**Figure S4**). Therefore, the simulation scenarios were capable of reproducing previously observed patterns, suggesting the appropriateness in the choice of parameter combinations.

# Empirical data

## Data selection criteria

**Hokkaido, Japan.** In Hokkaido, most data were collected from summer to fall. We screened data through the following procedure:

1.  We listed recorded fish species and re-organized species names to make them consistent across data sources. We removed the following species at this stage: (1) identified at the family-level; (2) marine fish species (including species that occasionally use brackish/freshwater habitats).
2.  We selected sampling sites based on the following criteria: (1) surveys were conducted with netting and/or electrofishing, (2) surveys were designed to collect a whole fish community, (3) sites contained reliable coordinates (sites with coordinates identical at 3 decimal degrees were treated as the same site), and (4) sites did not involve unidentified species (genus level) that are rarely observed in the data set (\< 100 sites occurrence).
3.  For sites with multiple visits (i.e., temporal replicates), we used the latest-year observation at each sampling site to minimize variation in sampling efforts among sites. Surveys that occurred in the same year were aggregated into a single observation.
4.  We confined sites to those with the latest observation year of $\ge$ 1990. Although the data set contained observations from 1953, we added this restriction to align the observation period with the data set in the Midwest, US.
5.  Four genera (*Lethenteron*, *Pungitius*, *Rhinogobius*, and *Tribolodon*) were treated as species groups (i.e., spp.) as their taxonomic resolutions varied greatly among data sources due to difficulties in identifying species.

**Midwest, US.** In the Midwest, the data set covered most of Upper Mississippi (Hydrologic Unit Code 2 [HUC 2] , region 07, as defined by U.S. Geological Survey and U.S. Department of Agriculture Natural Resources Conservation Service [@u.s.geologicalsurveyandu.s.departmentofagriculturenaturalresourcesconservationserviceFederalStandardsProcedures2013]) and the part of Great Lakes (HUC 2, region 04), Missouri (HUC 2, region 10), and Ohio (HUC 2, region 05). Fish data were collected from summer to fall with electrofishing (backpack, barge-type, or boat-mounted) and supplemental netting at some locations. We screened data through the following procedure:

1.  We used data of the Upper Mississippi (HUC 2, region 07) and Great Lakes basins (HUC 2, region 04) as most sites are included in these regions.
2.  We removed records of unidentified species, hybrid species, and commercial species apparently absent in the wild (e.g., goldfish).
3.  We used the latest observation at each sampling site to minimize variation in sampling efforts among sites.

## Asymptotic species richness

We evaluated sensitivity of asymptotic species richness (Chao 2 estimator) to sample size (i.e., the number of sampling sites in a watershed). We simulated presence-absence data of species with known values of true species richness $S_{true}$ and the number of sampling sites $N_{site}$. In this simulation, presence of species $i$ at site $x$, $J_{ix}$, was drawn from a Bernoulli distribution as $J_{ix} \sim Bernoulli(p_{i}\kappa_{ix})$ where $p_i$ is the detection probability for species $i$ and $\kappa_{ix}$ is the presence probability of species $i$ at site $x$. Based on the incidence frequency of simulated species $F_i = \sum_{x = 1}^{N_{site}}J_{ix}$, we estimated asymptotic species richness using the iNEXT function in the R package 'iNEXT' [@hsiehINEXTPackageRarefaction2016]. We calculated % bias of estimated asymptotic species richness $S_{est}$:

```{=tex}
\begin{equation}
  \%~bias = \frac{100(S_{est} - S_{true})}{S_{true}}
\end{equation}
```
Positive values of % bias indicate an overestimation of species richness, while negative values indicate an underestimation.

We used the following values for parameters: $S_{true} = 10, 40, 70, 100$ and $N_{site} = 5, 10, 15, 20, 100$. We produced 100 replicates of simulated data sets for each parameter combination, resulting in a total of 2000 simulation replicates. In each simulation replicate, the probabilities of detection and true presence were drawn randomly from uniform distributions as $p_i \sim Unif(0.3, 0.8)$ and $\kappa_{ix} \sim Unif(0, 1)$.

The number of sampling sites influenced the estimation accuracy of asymptotic species richness. The % bias decreased sharply as the number of sampling sites increased (**Figure S20**). In particular, the estimation bias with a small sample size ($N_{site} = 5$) was substantial when the true species richness $S_{true}$ was low; some estimates showed \> 150% bias (**Figure S20**). Given the simulation results, estimates of asymptotic species richness at watersheds with \< 10 sampling sites may involve substantial statistical uncertainty.

## Average predictive comparison

For regression models, standardized regression coefficients (or its variant) are perhaps the most common summary when comparing effect sizes of explanatory variables. These values are useful if they are directly interpretable. For example, in linear regression models without interactions and variable transformations, regression coefficients have direct interpretations as they represent additive effects. However, there are many cases where regression coefficients are difficult to interpret. In our case, we regressed the log-transformed species richness on explanatory variables, so regression coefficients do not have direct interpretations on the original scale of the response variable $y$ (i.e., species richness).

The average predictive comparison provides an intuitive yet rigorous way to quantify effect sizes for each of explanatory variables in regression models with interactions and/or variable transformations [@gelmanAveragePredictiveComparisons2007]. The basic predictive comparison $\delta_u$ is an expected change in $y$ on the original scale per a unit difference of the explanatory variable of interest:

```{=tex}
\begin{equation}
  \delta_u(u^{(1)} \to u^{(2)}, v, \Theta) = \frac{E(y|u^{(2)}, v, \Theta) - E(y|u^{(1)}, v, \Theta)}{u^{(2)} - u^{(1)}}
\end{equation}
```
where $E(\cdot)$ is a known function (e.g., exponential), $u$ the input of interest (a value for the explanatory variable of interest), $v$ all the other inputs (a vector in general), $\Theta$ a set of parameters in a regression model. In general, $\delta_u$ rests on $u^{(1)}$ and $u^{(2)}$ (the lower and higher points of the hypothesized change in the explanatory variable of interest), $v$, and $\Theta$. Therefore, Gelman and Pardoe [@gelmanAveragePredictiveComparisons2007] defined the *average predictive comparison* $\Delta_u$, the mean value of $\delta_u$ over specified distributions of the inputs and model parameters:

```{=tex}
\begin{equation}
  \Delta_u = \frac{\int\int_{u^{(2)}>u^{(1)}}du^{(1)}du^{(2)}\int dv \int d\Theta (E(y|u^{(2)}, v, \Theta) - E(y|u^{(1)}, v, \Theta))~p(u^{(1)}|v)~p(u^{(2)}|v)~p(v)~p(\Theta)}
                  { \int\int_{u^{(2)}>u^{(1)}}du^{(1)}du^{(2)}\int dv \int d\Theta~(u^{(2)} - u^{(1)})~ p(u^{(1)}|v)~p(u^{(2)}|v)~p(v)~p(\Theta) }
\end{equation}
```
However, directly using the above equation is impractical because estimating $p(u^{(1)}|v)$ and $p(u^{(2)}|v)$ from finite data points is challenging (especially when $v$ is continuous). Instead, we estimated $\Delta_u$ using the following equation proposed by Gelman and Pardoe [@gelmanAveragePredictiveComparisons2007]:

```{=tex}
\begin{equation}
  \hat\Delta_u = \frac{\sum_{i=1}^n \sum_{j=1}^n \sum_{s=1}^S w_{ij}(E(y|u_j, v_i, \Theta_s) - E(y|u_i, v_i, \Theta_s))sign(u_j - u_i)}
                      {\sum_{i=1}^n \sum_{j=1}^n \sum_{s=1}^S w_{ij}(u_j - u_i)sign(u_j - u_i)}
\end{equation}
```
The summations over $n$ data ($i$ and $j$ are a given data point) and $S$ parameter replicates ($s$ is a given simulation replicate) are a realization of averaging over the distributions of $(u^{(1)},v)$ , $u^{(2)}$, and $\Theta$. The factor $w_{ij}$ is a weight that serves to approximate $p(u^{(1)}|v)$ and $p(u^{(2)}|v)$. In theory, $v$ must be held constant while the input of interest changes from $u^{(1)}$ to $u^{(2)}$. However, there are, if any, few transitions from $u^{(1)}$ to $u^{(2)}$ with a common $v$. We approximate such exact transitions by assigning each pair of data points with a weight:

```{=tex}
\begin{equation}
  w_{ij} = w(v_i, v_j)
\end{equation}
```
The weight factor should represent the likelihood of $u$ changing from $u_i$ to $u_j$ with a common $v = v_i$ ($v_i$ is in general a vector of explanatory variables for data point $i$). We used the following weighting function based on Mahalanobis distances following Gelman and Pardoe [@gelmanAveragePredictiveComparisons2007]:

```{=tex}
\begin{equation}
w(v_i, v_j) = \frac{1}{1 + (v_i - v_j)^T\Omega_v^{-1}(v_i-v_j)}
\end{equation}
```
where $\Omega_v^{-1}$ is the inverse of the variance-covariance matrix of $v$. Note that $v=v_i$ when $u = u_i$, and $v=v_j$ when $u = u_j$. The function gives the maximum weight for a pair of $u_i$ and $u_j$ if $v_i = v_j$ while giving a less weight as the Mahalanobis distance between $v_i$ and $v_j$ increases. This property makes sense because our goal is to approximate the probability of transition from $u_i$ to $u_j$ with a common $v$ (i.e., $v_i = v_j$).

We estimated $\hat \Delta_u$ for watershed area and branching probability using the estimated coefficients $\hat \xi_k$ of the regression model explaining $\gamma$ diversity (see **Table 1**). A thousand of vectors of simulated parameters $\Theta_s$ were drawn from normal distributions with means of $\hat \xi_k$ and standard deviations of $\sigma_{\xi, k}$ (the estimated standard errors of parameters $\xi_k$). The estimated average predictive comparisons are $\hat \Delta_{u} = \frac{1}{S} \sum_{s=1}^S \hat \Delta_{u,s}$ where

```{=tex}
\begin{equation}
  \hat \Delta_{u,s} = \frac{\sum_{i=1}^n \sum_{j=1}^n w_{ij}(E(y|u_j, v_i, \Theta_s) - E(y|u_i, v_i, \Theta_s))sign(u_j - u_i)}
                           {\sum_{i=1}^n \sum_{j=1}^n w_{ij}(10^{u_j} - 10^{u_i})sign(u_j - u_i)}
\end{equation}
```
We exponentiated $u_i$ and $u_j$ in the denominator because we log-transformed watershed area and branching probability in the statistical model. Then

```{=tex}
\begin{equation}
  S.E.(\hat \Delta_{u,s}) = \sqrt{\frac{1}{S-1} \sum_{s=1}^S (\hat \Delta_{u,s} - \hat \Delta_{u})^2}
\end{equation}
```
The estimated average predictive comparisons were summarized in **Table S7**.

\newpage

# Tables

## Table S1 Simulation parameter (sensitivity analysis)

**Table S1** Parameter values used in the sensitivity analysis of the metacommunity simulation. See the main text for model details.

```{r chunk_table1}

  source(here::here('theory/table_sim_parameter_sensitivity.R'))
  table_output

```

\newpage

## Table S2 Simulation parameter (main analysis)

**Table S2** Values and interpretation of simulation parameters used in the main simulation. See the main text for model details.

```{r chunk_table2}

  source(here::here('theory/table_sim_parameter_main.R'))
  table_output

```

\newpage

## Table S3 Sensitivity analysis for ecosystem size effects

**Table S3** Sensitivity analysis of the metacommunity simulation. Parameter estimates of linear regression models (standard errors in parenthesis) are shown. Response variables are the effects of the number of habitat patches ($N_{p}$) on $\alpha$, $\beta$, and $\gamma$ diversity. Explanatory variables (i.e., simulation parameters) were standardized to a mean of zero and a standard deviation of one prior to the analysis. See **Tables S1 and S2** for interpretation of the simulation parameters.

```{r chunk_table3, results='asis'}
  
  cat(star_area, sep='\n')

```

\newpage

## Table S4 Sensitivity analysis for ecosystem complexity effects

**Table S4** Sensitivity analysis of the metacommunity simulation. Parameter estimates of linear regression models (standard errors in parenthesis) are shown. Response variables are the effects of branching probability ($P_{b}$) on $\alpha$, $\beta$, and $\gamma$ diversity. Explanatory variables (i.e., simulation parameters) were standardized to a mean of zero and a standard deviation of one prior to the analysis. See **Tables S1 and S2** for interpretation of the simulation parameters.

```{r chunk_table4, results='asis'}

  cat(star_bp, sep='\n')

```

\newpage

## Table S5 List of fish species in Hokkaido, Japan

**Table S5** List of fish species in Hokkaido, Japan, included in our statistical analysis. `r n_distinct(dat_hkd_site$Species)` species are ordered alphabetically, along with the number of sites present and % occupancy of `r n_distinct(dat_hkd_site$SiteID)` sites.

```{r chunk_table5}

  source(here::here('empirical/table_splist_hkd.R'))
  table_output

```

\newpage

## Table S6 List of fish species in Midwest, US

**Table S6** List of fish species in the Midwest, US, included in our statistical analysis. `r n_distinct(dat_mw_site$Species)` species are ordered alphabetically, along with the number of sites present and % occupancy of `r n_distinct(dat_mw_site$SiteID)` sites.

```{r chunk_table6}

  source(here::here('empirical/table_splist_mw.R'))
  table_output
  
```

\newpage

## Table S7 Average predictive comparisons

Estimated average predictive comparisons. Regression coefficients were derived from the regression model explaining $\gamma$ diversity (see **Table 1** in the maintext for the estimated regression parameters). The average predictive comparisons (an expected increase in $\gamma$ diversity per a unit difference of the explanatory variable of interest) were estimated based on units of 1000 km^2^ for watershed area and 0.1 for branching probability.

```{r chunk_table7}

  source(here::here('empirical/table_apcomp.R'))
  table_output

```

\newpage

# Figures

```{r chunk_fig_set1}

source(here::here("theory/figure_alpha_pattern.R"))

figure_cap <- conditional_text <- NULL

for(i in seq_len(nrow(param))){
  if(param$sigma_h[i] > param$sigma_l[i]){
    conditional_text[i] <- "In this simulation, environmental variation at headwaters ($\\sigma_{h}$) exceeds local environmental noise ($\\sigma_{l}$)."
  }else{
    if(param$sigma_h[i] == param$sigma_l[i]){
      conditional_text[i] <- "In this simulation, environmental variation at headwaters ($\\sigma_{h}$) is equal to local environmental noise ($\\sigma_{l}$)."
    }else{
      conditional_text[i] <- "In this simulation, environmental variation at headwaters ($\\sigma_{h}$) is less than local environmental noise ($\\sigma_{l}$)."
    }
  }
  
  figure_cap[i] <- paste0("Theoretical predictions for longitudinal gradients of local species richness in branching networks. The longitudinal position of each habitat patch (x-axis) was expressed as the number of upstream habitat patches. `r conditional_text[",i,"]` Lines and shades are loess curves fitted to simulated data and their 95% confidence intervals. Each panel represents different ecological scenarios under which metacommunity dynamics were simulated. Rows represent different competition strength. Competitive coefficients ($\\alpha_{ij}$) were varied randomly from 0 to 1.5 (top, strong competition) or 0.75 (bottom, weak competition). Columns and lines represent different dispersal scenarios (dispersal distance and probability). Left and right columns show long-distance (the rate parameter of an exponential dispersal kernel $\\theta = 0.10$) and short-distance dispersal ($\\theta = 1.0$) scenarios repectively. Red and blue lines show low ($p_d = 0.01$) and high dispersal probabilities ($p_d = 0.10$). Other parameters are as follows: environmental variation at headwaters $\\sigma_{h} = `r param$sigma_h[",i,"]`$; local environmental noise $\\sigma_{l} = `r param$sigma_l[",i,"]`$; ecosystem size $N_{p} = 100$; ecosystem complexity $P_{b} = 0.5$.")
}

figure_alpha_pattern <- NULL
for(i in seq_len(nrow(param))){
  j <- i
  knit_chunk <- paste0("## Figure S", j, " Longitudinal gradient of local species richness ($\\sigma_{h} = `r param$sigma_h[",i,"]`$, $\\sigma_{l} = `r param$sigma_l[",i,"]`$)", 
                       "\n```{r fig", j,", fig.width = 8, fig.height = 6}\n\n h[[",i,"]] \n\n```\n",
                       "\n **Figure S", j, "** ", figure_cap[i], "\n\\pagebreak\n")
  figure_alpha_pattern <- c(figure_alpha_pattern, knit_chunk)
}

```

`r paste(knit(text = figure_alpha_pattern), collapse = '\n')`

```{r chunk_fig_set2}

source(here::here("theory/figure_n_patch_si.R"))

figure_cap <- conditional_text <- NULL

for(i in seq_len(nrow(param))){
  if(param$sigma_h[i] > param$sigma_l[i]){
    conditional_text[i] <- "In this simulation, environmental variation at headwaters ($\\sigma_{h}$) exceeds local environmental noise ($\\sigma_{l}$)."
  }else{
    if(param$sigma_h[i] == param$sigma_l[i]){
      conditional_text[i] <- "In this simulation, environmental variation at headwaters ($\\sigma_{h}$) is equal to local environmental noise ($\\sigma_{l}$)."
    }else{
      conditional_text[i] <- "In this simulation, environmental variation at headwaters ($\\sigma_{h}$) is less than local environmental noise ($\\sigma_{l}$)."
    }
  }
  
  figure_cap[i] <- paste0("Theoretical predictions for ecosystem size influences (the number of habitat patches) on $\\alpha$, $\\beta$, and $\\gamma$ diversity in branching networks. `r conditional_text[",i,"]` Lines and shades are loess curves fitted to simulated data and their 95% confidence intervals. Each panel represents different ecological scenarios under which metacommunity dynamics were simulated. Rows represent different competition strength. Competitive coefficients ($\\alpha_{ij}$) were varied randomly from 0 to 1.5 (top, strong competition) or 0.75 (bottom, weak competition). Columns represent different dispersal scenarios. Two dispersal parameters were chosen to simulate scenarios with long-distance (the rate parameter of an exponential dispersal kernel $\\theta = 0.10$) and short-distance dispersal ($\\theta = 1.0$). Other parameters are as follows: dispersal probability $p_{d} = `r param$p_d[",i,"]`$; environmental variation at headwaters $\\sigma_{h} = `r param$sigma_h[",i,"]`$; local environmental noise $\\sigma_{l} = `r param$sigma_l[",i,"]`$.")
}

figure_size <- NULL
for(i in seq_len(nrow(param))){
  j <- i + 4
  knit_chunk <- paste0("## Figure S", j, " Influence of ecosystem size ($p_{d} = `r param$p_d[",i,"]`$, $\\sigma_{h} = `r param$sigma_h[",i,"]`$, $\\sigma_{l} = `r param$sigma_l[",i,"]`$)", 
                       "\n```{r fig", j,", fig.width = 8, fig.height = 6}\n\n f[[",i,"]] \n\n```\n",
                       "\n **Figure S", j, "** ", figure_cap[i], "\n\\pagebreak\n")
  figure_size <- c(figure_size, knit_chunk)
}

```

`r paste(knit(text = figure_size), collapse = '\n')`

```{r chunk_fig_set3}

source(here::here("theory/figure_p_branch_si.R"))

figure_cap <- conditional_text <- NULL

for(i in seq_len(nrow(param))){
  if(param$sigma_h[i] > param$sigma_l[i]){
    conditional_text[i] <- "In this simulation, environmental variation at headwaters ($\\sigma_{h}$) exceeds local environmental noise ($\\sigma_{l}$)."
  }else{
    if(param$sigma_h[i] == param$sigma_l[i]){
      conditional_text[i] <- "In this simulation, environmental variation at headwaters ($\\sigma_{h}$) is equal to local environmental noise ($\\sigma_{l}$)."
    }else{
      conditional_text[i] <- "In this simulation, environmental variation at headwaters ($\\sigma_{h}$) is less than local environmental noise ($\\sigma_{l}$)."
    }
  }
  
  figure_cap[i] <- paste0("Theoretical predictions for ecosystem complexity influences (branching probability) on $\\alpha$, $\\beta$, and $\\gamma$ diversity in branching networks. `r conditional_text[",i,"]` Lines and shades are loess curves fitted to simulated data and their 95% confidence intervals. Each panel represents different ecological scenarios under which metacommunity dynamics were simulated. Rows represent different competition strength. Competitive coefficients ($\\alpha_{ij}$) were varied randomly from 0 to 1.5 (top, strong competition) or 0.75 (bottom, weak competition). Columns represent different dispersal scenarios. Two dispersal parameters were chosen to simulate scenarios with long-distance (the rate parameter of an exponential dispersal kernel $\\theta = 0.10$) and short-distance dispersal ($\\theta = 1.0$). Other parameters are as follows: dispersal probability $p_{d} = `r param$p_d[",i,"]`$; environmental variation at headwaters $\\sigma_{h} = `r param$sigma_h[",i,"]`$; local environmental noise $\\sigma_{l} = `r param$sigma_l[",i,"]`$.")
}

figure_complex <- NULL
for(i in seq_len(nrow(param))){
  j <- i + 11
  knit_chunk <- paste0("## Figure S", j, " Influence of ecosystem complexity ($p_{d} = `r param$p_d[",i,"]`$, $\\sigma_{h} = `r param$sigma_h[",i,"]`$, $\\sigma_{l} = `r param$sigma_l[",i,"]`$)", 
                       "\n```{r fig", j,", fig.width = 8, fig.height = 6}\n\n g[[",i,"]] \n\n```\n",
                       "\n **Figure S", j, "** ", figure_cap[i], "\n\\pagebreak\n")
  figure_complex <- c(figure_complex, knit_chunk)
}

```

`r paste(knit(text = figure_complex), collapse = '\n')`

## Figure S19 Correlation structure of explanatory variables

```{r fig_s19, fig.height = 10, fig.width = 10}

source(here::here('empirical/figure_correlation.R'))

```

**Figure S19** Correlation structure of potential explanatory variables for riverine diversity metrics. Numeric values in each cell are the Pearson's correlation coefficients. Positive and negative correlations were colored in blue and red, respectively, and darker colors indicate stronger correlations. Environmental variables (temperature, precipitation, elevation, logit fraction of forest, logit fraction of urban, logit fraction of agriculture, and dam density) were expressed as deviations from the regional averages to remove any regional effects.

\newpage

## Figure S20 Sensitivity analysis of asymptotic species richness

```{r fig_s20, fig.height = 6, fig.width = 6}

source(here::here('empirical/figure_iNEXT_sim.R'))

```

**Figure S20** Sensitivity analysis of asymptotic species richness in relation to true species richness $S_{true}$ (panels) and the number of sampling sites $N_{site}$ (x-axis). Positive values of % bias indicate an overestimation of species richness, while negative values indicate an underestimation. Different panels show results with different true species richness. The center lines are median values, the box boundaries 25 and 75 percentiles, and the whiskers 5 and 95 percentiles. Dots represent individual simulation replicates.

\newpage

# References
